lib/faraday/request/part_params.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module Faraday
+  class Request
+    # Allows multipart posts while specifying headers of a part
+    class PartParams
```

Comment:
I like the name `ParamsPart` instead, as it describes what kind of Part it is. I could even see us renaming `UploadIO` to something like `FilePart` for consistency (while keeping the old class around for backwards compatibility, of course).

Reply:
I didn't do anything about this, but I agree `FilePart` would be a good name.

Reply:
I miss-read your comment. I changed the name to `ParamsPart`.